CI/wininstaller/installer.iss

```diff
@@ -0,0 +1,733 @@
+              //MsgBox('Failed to copy file: ' + SourceFile + ' to ' + DestFile, mbError, MB_OK);
+          end;
+        end
+        else if (FindRec.Name <> '.') and (FindRec.Name <> '..') then
+        begin
+          // Copy subdirectories recursively
+          CopyFolderContents(SourceFile, DestFile, Overwrite);
+        end;
+      until not FindNext(FindRec);
+    finally
+      FindClose(FindRec);
+    end;
+  end
+  //else
+  //  MsgBox('No files found in directory: ' + SourceDir, mbError, MB_OK);
+end;
+
+
+// A huge workaround to get non-admin profile name on elevated installer as admin
+function WTSQuerySessionInformation(hServer: THandle; SessionId: Cardinal; WTSInfoClass: Integer; var pBuffer: DWord; var BytesReturned: DWord): Boolean;
+  external 'WTSQuerySessionInformationW@wtsapi32.dll stdcall';
+
+procedure WTSFreeMemory(pMemory: DWord);
+  external 'WTSFreeMemory@wtsapi32.dll stdcall';
+
+procedure RtlMoveMemoryAsString(Dest: string; Source: DWord; Len: Integer);
+  external 'RtlMoveMemory@kernel32.dll stdcall';
+
+const
+  WTS_CURRENT_SERVER_HANDLE = 0;
+  WTS_CURRENT_SESSION = -1;
+  WTSUserName = 5;
+
+function GetCurrentSessionUserName: string;
+var
+  Buffer: DWord;
+  BytesReturned: DWord;
+  QueryResult: Boolean;
+begin
+  // Initialize Result to an empty string
+  Result := '';
+
+  // Query the username for the current session
+  QueryResult := WTSQuerySessionInformation(
+    WTS_CURRENT_SERVER_HANDLE, WTS_CURRENT_SESSION, WTSUserName, Buffer, BytesReturned);
+
+  if not QueryResult then
+  begin
+    // Error if the query fails
+    Exit;
+  end;
+
+  try
+    // Set the length of the result string (BytesReturned includes null terminator)
+    SetLength(Result, (BytesReturned div 2) - 1); // Divide by 2 for Unicode and exclude null terminator
+
+    // Copy the buffer contents into the result string
+    RtlMoveMemoryAsString(Result, Buffer, BytesReturned - 2); // Exclude null terminator
+  finally
+    // Free the allocated memory
+    WTSFreeMemory(Buffer);
+  end;
+
+end;
+
+
+function GetCommonProgramFilesDir: String;
+begin
+  // Check if the installer is running on a 64-bit system
+  if DirExists(ExpandConstant('{win}\syswow64')) then
+  begin
+    if ExpandConstant('{#InstallerArch}') = 'x64' then
+      // For 64-bit installer, return the 64-bit Program Files directory
+      Result := ExpandConstant('{commonpf64}')
+    else
+      // For 32-bit installer on 64-bit system, return the 32-bit Program Files directory
+      Result := ExpandConstant('{commonpf32}');
+  end
+  else
+    // On 32-bit systems, always return the 32-bit Program Files directory
+    Result := ExpandConstant('{commonpf32}');
+end;
+
+
+function GetDefaultDir(Default: String): String;
+begin
+  if IsAdmin then
+    // Default to Program Files for admins
+    Result := GetCommonProgramFilesDir + '\VCMI'
+  else
+    // Default to User AppData for non-admin users
+    Result := GlobalUserAppdataFolder + '\VCMI';
```

Comment:
would `C:\Games\` be a better choice perhaps?

Reply:
User can always check it. Also not sure if in non-admin environment will be this allowed without addition system checks or prompts. Also installing to system drive root becomes not only user profile installation. Normally will users install in same way to Program Files as usual